src/pybind/mgr/dashboard_v2/models/pool.py

```diff
@@ -0,0 +1,149 @@
+    flags = JsonField(base_type=list, blank=True, default=[])
+
+    _compr_mode_choices = 'force aggressive passive none'.split(' ')
+    _compr_algorithm_choices = 'none snappy zlib zstd lz4'.split(' ')
+    compression_mode = CharField(default='none', choices=[(x, x) for x in _compr_mode_choices])
+
+    compression_algorithm = CharField(default='none',
+                                      choices=[(x, x) for x in _compr_algorithm_choices])
+    # TODO: Add validation for value to be between 0 and 1
+    compression_required_ratio = FloatField(default=0.0)
+    compression_max_blob_size = IntegerField(default=0)
+    compression_min_blob_size = IntegerField(default=0)
+
+    pool_snaps = JsonField(base_type=list, editable=False, blank=True)
+
+    application_metadata = JsonField(base_type=dict, default={}, blank=True)
+
+    @staticmethod
+    def get_all_objects(api_controller, query):
+        """":type api_controller: RESTController"""
+        result = []
+
+        osd_map = api_controller.mgr.get('osd_map')
+
+        pool_data = None
+
+        for pool_data in osd_map['pools']:
+
+            if sys.version_info < (3, 0):
+                for key, value in pool_data.items():
+                    if isinstance(value, long):
+                        pool_data[key] = int(value)
+
+            def options(key, default=None):
+                opts = pool_data.get('options', {})
+                return opts.get(key, default)
+
+            object_data = {
+                'id': pool_data['pool'],
+                'name': pool_data['pool_name'],
+                'type': {1: 'replicated', 3: 'erasure'}[pool_data['type']],  # type is an
+                                                                             # undocumented dump of
+                # https://github.com/ceph/ceph/blob/289c10c9c79c46f7a29b5d2135e3e4302ac378b0/src/osd/osd_types.h#L1035
+                'erasure_code_profile_id':
+                    (pool_data['erasure_code_profile'] if 'erasure_code_profile' in pool_data
+                     else None),
+                'last_change': pool_data['last_change'],
+                'full': 'full' in pool_data['flags_names'],
+                'min_size': pool_data['min_size'],
+                # TODO: 'crash_replay_interval': pool_data['crash_replay_interval'],
+                'pg_num': pool_data['pg_num'],
+                'size': pool_data['size'],
+                'crush_ruleset': pool_data['crush_rule'],
+                # Considered advanced options
+                'pgp_num': pool_data['pg_placement_num'],
+                'stripe_width': pool_data['stripe_width'],
+                'quota_max_bytes': pool_data['quota_max_bytes'],
+                'quota_max_objects': pool_data['quota_max_objects'],
+                # Cache tiering related
+                'cache_mode': pool_data['cache_mode'],
+                'tier_of_id': pool_data['tier_of'] if pool_data['tier_of'] > 0 else None,
```

Comment:
`s/tier_of_id/tier_of/`. The `_id` suffix doesn't work right now, as we now have IntegerFields. Dito for `write_tier_id` and `read_tier_id`